cli/internal/graph/graph.go

```diff
@@ -86,3 +87,106 @@ func getTaskFromPipeline(pipeline fs.Pipeline, taskID string, taskName string) (
 
 	return &taskDefinition, nil
 }
+
+func (g *CompleteGraph) getMergedTaskDefinition(pkg *fs.PackageJSON, taskID string, taskName string) (*fs.TaskDefinition, error) {
+	taskDefinitions, err := g.getTaskDefinitionChain(pkg, taskID, taskName)
+	if err != nil {
+		return nil, err
+	}
+
+	// reverse the array, because we want to start with the end of the chain.
+	for i, j := 0, len(taskDefinitions)-1; i < j; i, j = i+1, j-1 {
+		taskDefinitions[i], taskDefinitions[j] = taskDefinitions[j], taskDefinitions[i]
+	}
+
+	// Start with an empty definition
+	mergedTaskDefinition := &fs.TaskDefinition{}
+
+	// For each of the TaskDefinitions we know of, merge them in
+	for _, taskDef := range taskDefinitions {
+		mergedTaskDefinition.Outputs = taskDef.Outputs
```

Comment:
is this not just essentially assigning `mergedTaskDefinition` to `taskDefinitions[-1]`? Each iteration overwrites (unless the assignment is doing the merge) and all but the last operation are meaningless

Reply:
Yeah I should be checking if that key exists has a non-null value before assigning, good catch 🤦🏾. Might run into some issues here differentiating between values that need to override vs missing keys, since I think the json unmarshaled struct doesn't differentiate between those 😢

Reply:
actually I take that back... I cannot write a test case for this, because we only allow a taskDefinition chain of two right now, so a single overwrite "works". It is faulty code, but when we enable extending from other workspaces, and thus allowing long chains, the test case will reveal the fault. 

Is it ok to leave this in right now, since I cannot write a test for it? I will add a code comment

Reply:
never mind, it is possible to exercise the fault. Adding a test case now.

Reply:
hmm writing this test has revealed a show stopper bug. The place we are resolving this configs are _after_ the taskGraph has already been created. Since `dependsOn` can change the graph, it's too late by this point (unless we also go an change that graph here). I think that is not safe since execution has already begun in parallel. 

I need to revisit this.